kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaConsumerInput.java

```diff
@@ -101,4 +112,29 @@ private void commitOffsets( Map.Entry<List<List<Object>>, Result> rowsAndResult
     ( (KafkaStreamSource) source ).commitOffsets( rowsAndResult.getKey() );
   }
 
+  private boolean checkKafkaConnectionStatus( KafkaConsumerInputMeta meta ) {
+    boolean kafkaConnectionStatus = false;
+    Properties props = new Properties();
+    props.put( ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, meta.getBootstrapServers()  );
+    props.put( ConsumerConfig.GROUP_ID_CONFIG,  meta.getConsumerGroup()  );
+    props.put( ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class );
+    props.put( ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class );
+    meta.getConfig().entrySet()
+      .forEach( ( entry -> props.put( entry.getKey(),
+        (String) entry.getValue() ) )  );
+
+    AdminClient client = AdminClient.create(props);
+    Collection<Node> nodes = null;
+    try {
+      nodes = client.describeCluster().nodes().get();
+    }catch( ExecutionException e ) {
+      logError( BaseMessages.getString( PKG, "KafkaConsumerInput.Error.WaitingForConsumerToConnect" ) );
```

Comment:
Why are we ignoring the exception being thrown here?  Could there be any useful information in it that we'd want shown in the log?

Reply:
This code is related to handle TimeOutException from Kafka Broker  and showing this error message in PDI logs .
As this code only identifies  TimeOutException  , So we are not throwing the exception .
Based on the flag we are able to stop the execution of PDI whenever TimeOutException Occurs .